libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+   * @param modifiedTime - The UNIX epoch time when this was last modified.
+   */
+  public static updateTimes(path: string, accessedTime: number, modifiedTime: number): void {
+    fsx.utimes(path, accessedTime, modifiedTime);
+  }
+
+  /**
+   * Changes the permissions mode of a file, folder, or link.
+   * Behind the scenes it uses `fs.chmodSync()`.
+   * @param path - The absolute or relative path to the object that should be updated.
+   * @param mode - should be a UNIX-style permissions modifier number (e.g. 777 or 666 etc)
+   */
+  public static changeMode(path: string, mode: number): void {
+    fsx.chmodSync(path, mode);
+  }
+
+  /**
+   * Moves a file. The folder must exist, unless the `ensureFolder` option is provided.
+   * Behind the scenes it uses `fsx.moveSync()`
+   * @param sourcePath - The absolute or relative path to the source file.
+   * @param destinationPath - The absolute or relative path where the file should be moved to.
+   * @param options - Optional settings that can change the behavior. Type: `IMoveOptions`
+   */
+  public static move(sourcePath: string, destinationPath: string, options?: IMoveOptions): void {
+    options = {
+      overwrite: false,
+      ensureFolder: false,
+      ...options
+    };
+
+    if (options.ensureFolder) {
+      FileSystem.createFolder(pathUtilities.basename(sourcePath));
+    }
+    fsx.moveSync(sourcePath, destinationPath, { overwrite: options.overwrite });
+  }
+
+  // ===============
+  // FOLDER OPERATIONS
+  // ===============
+
+  /**
+   * Recursively creates a folder at a given path.
+   * Behind the scenes is uses `fsx.ensureDirSync()`.
+   * @param folderPath - The absolute or relative path of the folder which should be created.
+   */
+  public static createFolder(folderPath: string): void {
+    fsx.ensureDirSync(folderPath);
+  }
+
+  /**
+   * Reads the contents of the folder, not including "." or "..".
+   * Behind the scenes it uses `fs.readdirSync()`.
+   * @param folderPath - The absolute or relative path to the folder which should be read.
+   * @param options - Optional settings that can change the behavior. Type: `IReadFolderOptions`
+   */
+  public static readFolder(folderPath: string, options?: IReadFolderOptions): Array<string> {
+    options = {
+      absolutePath: false,
+      ...options
+    };
+
+    if (!FileSystem.exists(folderPath)) {
+      throw new Error(`Cannot read contents of folder, as it does not exist: "${folderPath}"`);
+    }
+
+    const fileNames: Array<string> = fsx.readdirSync(folderPath);
+
+    if (options.absolutePath) {
+      return fileNames.map(fileName => pathUtilities.resolve(folderPath, fileName));
+    }
+
+    return fileNames;
+  }
+
+  /**
+   * Deletes a folder, including all of its contents.
+   * Behind the scenes is uses `fsx.removeSync()`
+   * @param folderPath - The absolute or relative path to the folder which should be deleted.
+   */
+  public static deleteFolder(folderPath: string): void {
```

Comment:
>deleteFolder [](start = 16, length = 12)

Document whether it is an error if the folder already does not exist.  (Hopefully not)